ast/Trees.cc

```diff
@@ -107,10 +108,7 @@ string ExpressionPtr::toStringWithTabs(const core::GlobalState &gs, int tabs) co
 }
 
 bool ExpressionPtr::isSelfReference() const {
-    if (auto local = cast_tree<Local>(*this)) {
-        return local->localVariable == core::LocalVariable::selfVariable();
-    }
-    return false;
+    return isa_tree<Self>(*this);
```

Comment:
I'm fine removing `.isSelfReference` and just inlining `isa_tree<Self>` everywhere. We only added it because when we started representing `self` with Local nodes.